packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -674,11 +687,34 @@ export class RichTextEditor extends RichText implements ErrorPattern {
         );
     }
 
-    // This method restore the cursor selection after setting the editor content when the editor is focused
+    private resetEditorButtonsState(
```

Comment:
Instead of telling tip tap what we think the button state should be based on its previous state, is there a way to ask tip tap to recompute the state? That seems like it could be more reliable to always leave tip tap in control rather than risking that our guess about the previous state and our guess about tip tap's current state are correct.

I don't know what's possible so I'm approving the current approach since you have tests that would fail if those guesses are wrong, but it would be good to see if it can be simplified.

Reply:
I don't think so, in one code path we are completely destroying and recreating the editor state. So anything we want to preserver I think we need to hold on to. Even the less destructive code path "refreshMarkdownContent" is saving and restoring the selection state manually..

Reply:
Similar to the selection state, tip tap doesn't know the previous states, So explicitly setting the cursor and button states whenever the `setMarkdown()` gets called.